pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -0,0 +1,195 @@
+}
+
+// Reconcile reads that state of the cluster for a UserSignup object and makes changes based on the state read
+// and what is in the UserSignup.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling UserSignup")
+
+	// Fetch the UserSignup instance
+	instance := &toolchainv1alpha1.UserSignup{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Read the status conditions and assign some variables
+	provisioning := condition.IsConditionTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupProvisioning)
+	completed := condition.IsConditionTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
+
+	// If the status is not yet provisioning or completed, check for automatic or manual approval
+	if !provisioning && !completed {
+		userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+		// then proceed with the signup
+		if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+			var targetCluster string
+
+			// If a target cluster hasn't been selected, select one from the members
+			if instance.Spec.TargetCluster != "" {
+				targetCluster = instance.Spec.TargetCluster
+			} else {
+				// Automatic cluster selection
+				members := cluster.GetMemberClusters()
+				if len(members) > 0 {
+					targetCluster = members[0].Name
+				}
+    			// TODO how do we handle the situation where no members are available?
+
+			}
+
+			// Provision the MasterUserRecord
+			userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+				{
+					TargetCluster: targetCluster,
+				},
+			}
+
+			mur := &toolchainv1alpha1.MasterUserRecord{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: instance.Spec.UserID,
+					Namespace: config.GetOperatorNamespace(),
+				},
+				Spec: toolchainv1alpha1.MasterUserRecordSpec{
+					UserID: instance.Spec.UserID,
+					UserAccounts: userAccounts,
+				},
+			}
+
+			err := r.client.Create(context.TODO(), mur)
+			if err != nil {
+				// Error creating the MasterUserRecord - requeue the request.
```

Comment:
`requeue the request` -> so the `Reconcile` method will be called again, correct?

Reply:
I believe so, I'm not actually sure what circumstances would cause an error here.